src/main/java/com/chromediopside/service/ProfileService.java

```diff
@@ -0,0 +1,77 @@
+package com.chromediopside.service;
+
+import com.chromediopside.model.GiTinderProfile;
+import com.chromediopside.model.Language;
+import com.chromediopside.repository.LanguageRepository;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.eclipse.egit.github.core.Repository;
+import org.eclipse.egit.github.core.User;
+import org.eclipse.egit.github.core.client.GitHubClient;
+import org.eclipse.egit.github.core.service.RepositoryService;
+import org.eclipse.egit.github.core.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class ProfileService {
+
+  @Autowired
+  private LanguageRepository languageRepository;
+
+  private static final String GET_REQUEST_IOERROR = "Some GitHub data not available for this accessToken!";
+
+  public List<GiTinderProfile> randomTenProfileByLanguage(String languageName) {
+    Language selectingLanguage = languageRepository.findOne(languageName);
+    Set<GiTinderProfile> profilesSet = selectingLanguage.getProfileSet();
```

Comment:
Profiles shouldn't be a part of a Language. It doesn't belong there.
You should create a custom profile db query to get the 10 random profiles by language.